agent/wptdriver/web_browser.cc

```diff
@@ -232,7 +232,12 @@ bool WebBrowser::RunAndWait() {
           ret = true;
           _status.Set(_T("Waiting up to %d seconds for the test to complete"), 
                       (_test._test_timeout / SECONDS_TO_MS) * 2);
-          DWORD wait_time = _test._test_timeout * 2;
+          DWORD wait_time;
+          if (_test._overall_timeout_in_ms != 0 &&
+            _test._overall_timeout_in_ms < _test._test_timeout * 2)
+            wait_time = _test._overall_timeout_in_ms; //Strict limit on this value
+          else
+            wait_time = _test._test_timeout * 2;

```

Comment:
The stricter this limit is, the better. We want WPT to run to as close to the timeout value  as possible. (obviously this isn't going to be perfect as there's a boot-up time for the browser and shutdown time). There's also a 10 second wait time after this statement for the browser process to finish, so I think that's more than adequate to let the browser finish saving/shutdown.